hack/test-go.sh

```diff
@@ -136,56 +138,58 @@ if [[ $iterations -gt 1 ]]; then
   fi
 fi
 
-cover_report_dir=""
-combined_cover_profile=""
-if [[ -n "${KUBE_COVER}" ]]; then
-  cover_report_dir="/tmp/k8s_coverage/$(kube::util::sortable_date)"
-  combined_cover_profile="${cover_report_dir}/combined-coverage.out"
-  kube::log::status "Saving coverage output in '${cover_report_dir}'"
-  mkdir -p ${cover_report_dir}
-  # The combined coverage profile needs to start with a line indicating which
-  # coverage mode was used (set, count, or atomic). This line is included in
-  # each of the coverage profiles generated when running 'go test -cover', but
-  # we strip these lines out when combining so that there's only one.
-  echo "mode: ${KUBE_COVERMODE}" >${combined_cover_profile}
-fi
-
 if [[ -n "${1-}" ]]; then
   test_dirs=$@
 else
   test_dirs=$(kube::test::find_dirs)
 fi
 
-# Run all specified tests, optionally collecting coverage if KUBE_COVER is set.
-for arg in ${test_dirs}; do
-  trap 'exit 1' SIGINT
-  pkg=${KUBE_GO_PACKAGE}/${arg}
-
-  cover_params=()
-  cover_profile=""
-  if [[ -n "${KUBE_COVER}" ]]; then
-    cover_profile=${cover_report_dir}/${arg}/coverage.out
-    mkdir -p "${cover_report_dir}/${arg}"
-    cover_params=(-cover -covermode="${KUBE_COVERMODE}" -coverprofile="${cover_profile}")
-  fi
-
-  go test "${goflags[@]:+${goflags[@]}}" \
-      ${KUBE_RACE} \
-      ${KUBE_TIMEOUT} \
-      "${cover_params[@]+${cover_params[@]}}" \
-      "${pkg}"
-  if [[ -f "${cover_profile}" ]]; then
-    # Include all coverage reach data in the combined profile, but exclude the
-    # 'mode' lines, as there should be only one.
-    grep -h -v "^mode:" ${cover_profile} >>${combined_cover_profile} || true
-  fi
-done
+# If we're not collecting coverage, run all requested tests with one 'go test'
+# command, which is much faster.
+if [[ -z "${KUBE_COVER}" ]]; then
+  kube::log::status "Running unit tests without code coverage"
+  echo "${test_dirs}" | xargs -n1 printf "${KUBE_GO_PACKAGE}/%s\n" | \
+      xargs go test -p=${KUBE_PROCS} "${goflags[@]:+${goflags[@]}}" \
```

Comment:
`xargs` is probably not helping here since you end up with a single command line anyways... Might as well just expand the array here (it's unlikely it will exceed the maximum number of arguments that the shell can take.)